eskipfile/remote.go

```diff
@@ -0,0 +1,146 @@
+package eskipfile
+
+import (
+	"errors"
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/routing"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type remoteEskipFile struct {
+	preloaded       bool
+	remotePath      string
+	localPath       string
+	eskipFileClient *WatchClient
+	threshold       int
+	verbose         bool
+}
+
+// RemoteWatch creates a route configuration client with (remote) file watching. Watch doesn't follow file system nodes,
+// it always reads (or re-downloads) from the file identified by the initially provided file name.
+func RemoteWatch(rf string, th int, v bool) (routing.DataClient, error) {
+	if !isFileRemote(rf) {
+		return Watch(rf), nil
+	}
+
+	tempFilename, err := ioutil.TempFile("", "routes")
```

Comment:
I see some potential in the usage of the tempfile, e.g. it could be used to fetch from the remote less frequently or with varying frequency (like slowing down when there are no changes), than how often the routing mechanism calls the LoadUpdate method. However, in the current implementation, I don't see this yet. Which is fine. What I don't fully understand why do we use the tempfile here? Can you give a short explanation? I may miss something.

Reply:
We're using a tempfile because this implementation relies 100% on the existing `Watch` data client, it's only a wrapper around it. When downloading the file, the data remains in memory, so we need it to dump it into the local file so Watch works properly.

Am I missing something?